radix-engine/src/updates/anemone.rs

```diff
@@ -0,0 +1,479 @@
+use super::*;
+use crate::blueprints::consensus_manager::*;
+use crate::blueprints::models::KeyValueEntryContentSource;
+use crate::blueprints::package::*;
+use crate::blueprints::pool::v1::constants::*;
+use crate::system::system_db_reader::*;
+use crate::track::*;
+use crate::vm::*;
+use sbor::{generate_full_schema, TypeAggregator};
+
+#[derive(Clone)]
+pub struct AnemoneSettings {
+    /// Changes the cost associated with validator creation.
+    pub validator_fee_fix: UpdateSetting<()>,
+
+    /// Exposes second-precision timestamp.
+    pub seconds_precision: UpdateSetting<()>,
+
+    /// Introduces BLS12-381 and Keccak-256 features.
+    pub vm_boot_to_enable_bls128_and_keccak256: UpdateSetting<()>,
+
+    /// Increases the math precision with native pool implementations.
+    pub pools_update: UpdateSetting<()>,
+}
+
+impl UpdateSettings for AnemoneSettings {
+    type BatchGenerator = AnemoneBatchGenerator;
+
+    fn all_enabled_as_default_for_network(network: &NetworkDefinition) -> Self {
+        Self {
+            validator_fee_fix: UpdateSetting::enabled_as_default_for_network(network),
+            seconds_precision: UpdateSetting::enabled_as_default_for_network(network),
+            vm_boot_to_enable_bls128_and_keccak256: UpdateSetting::enabled_as_default_for_network(
+                network,
+            ),
+            pools_update: UpdateSetting::enabled_as_default_for_network(network),
+        }
+    }
+
+    fn all_disabled() -> Self {
+        Self {
+            validator_fee_fix: UpdateSetting::Disabled,
+            seconds_precision: UpdateSetting::Disabled,
+            vm_boot_to_enable_bls128_and_keccak256: UpdateSetting::Disabled,
+            pools_update: UpdateSetting::Disabled,
+        }
+    }
+
+    fn create_batch_generator(&self) -> Self::BatchGenerator {
+        AnemoneBatchGenerator {
+            settings: self.clone(),
+        }
+    }
+}
+
+#[derive(Clone)]
+pub struct AnemoneBatchGenerator {
+    settings: AnemoneSettings,
+}
+
+impl ProtocolUpdateBatchGenerator for AnemoneBatchGenerator {
+    fn generate_batch(
+        &self,
+        store: &dyn SubstateDatabase,
+        batch_index: u32,
+    ) -> Option<ProtocolUpdateBatch> {
+        match batch_index {
+            // Just a single batch for Anemone, perhaps in future updates we should have separate batches for each update?
```

Comment:
Again, would rather remove any logic that's not needed.

Reply:
The batch index is to align with the node interface, but agree it could be improved

Reply:
Sounds good to me.